Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/FunctionBlock.java

```diff
@@ -0,0 +1,257 @@
+ * function together with it's arguments and visual representation.
+ */
+public class FunctionBlock extends Block implements InputBlock, OutputBlock {
+    /** The OutputAnchor of this FunctionBlock. */
+    private OutputAnchor output;
+
+    /** The function name. */
+    private StringProperty name;
+
+    /** The space containing the input anchor(s). */
+    @FXML private Pane inputSpace;
+
+    /** The space containing the output anchor. */
+    @FXML private BorderPane outputSpace;
+
+    /** The space containing all the arguments of the function. */
+    private ArgumentSpace argumentSpace;
+    
+    /** The space in which to nest the FunctionBlock's inner parts. */
+    @FXML private Pane nestSpace;
+    
+    /** The space in which the information of the function is displayed. */
+    @FXML private Pane functionInfo;
+    
+    /**
+     * Method that creates a newInstance of this class along with it's visual
+     * representation
+     *
+     * @param name
+     *            The name of the function.
+     * @param type
+     *            The function's type (usually a FuncT).
+     * @param pane
+     *            The parent pane in which this FunctionBlock exists.
+     */
+    public FunctionBlock(String name, Type type, CustomUIPane pane) {
+        super(pane);
+        this.name = new SimpleStringProperty(name);
+
+        this.loadFXML("FunctionBlock");
+
+        // Collect argument types
+        ArrayList<String> args = new ArrayList<>();
+        Type t = type;
+        while (t instanceof FuncT) {
+            FuncT ft = (FuncT) t;
+            args.add(ft.getArgs()[0].toHaskellType());
+            t = ft.getArgs()[1];
+        }
+        
+        List<Type> inputSignatures = new ArrayList<>();
+        Type functionSignature = getFunctionSignature();
+        for (int i = 0; i < args.size(); i++) {
+            if (functionSignature instanceof FuncT) {
+                Type inputSignature = BackendUtils.dive((FuncT) functionSignature, i + 1);
+                inputSignatures.add(inputSignature);
+            } else {
+                throw new FunctionDefinitionException();
+            }
+        }
+        
+        argumentSpace = new ArgumentSpace(this, inputSignatures);
+        argumentSpace.setKnotIndex(getAllInputs().size());
+        
+        nestSpace.getChildren().add(argumentSpace);
+        argumentSpace.knotIndexProperty().addListener(e -> invalidateKnotIndex());
+        
+        // Create an anchor for the result
+        output = new OutputAnchor(this);
+        outputSpace.setCenter(output);
+        
+        // Make sure the prefWidth is correctly updated.
+        this.prefWidthProperty().bind(functionInfo.widthProperty().add(argumentSpace.prefWidthProperty()));
+        
+        
+        this.setMinSize(Region.USE_PREF_SIZE, Region.USE_PREF_SIZE);
+        this.setMaxSize(Region.USE_PREF_SIZE, Region.USE_PREF_SIZE);
+        functionInfo.setMinWidth(Region.USE_PREF_SIZE);
```

Comment:
@i3anaan  This is where your resizing fails, if you print out the values for prefWidth it gives you the value 60, that is exactly the value that the functionBlock is being sized to.